api/src/main/java/io/cloudevents/CloudEvent.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2018 The CloudEvents Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cloudevents;
+
+import java.net.URI;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * An abstract event envelope, representing the 0.1 version of the <a href="https://github.com/cloudevents/spec/blob/master/spec.md">CNCF CloudEvent spec</a>.
+ *
+ */
+public interface CloudEvent<T> {
+    /**
+     * Type of occurrence which has happened. Often this property is used for routing, observability, policy enforcement, etc.
+     */
+    String getEventType();
+
+    /**
+     * The version of the CloudEvents specification which the event uses. This enables the interpretation of the context.
+     */
+    String getCloudEventsVersion();
+
+    /**
+     * This describes the event producer. Often this will include information such as the type of the event source, the organization publishing the event, and some unique identifiers.
+     * The exact syntax and semantics behind the data encoded in the URI is event producer defined.
+     */
+    URI getSource();
+
+    /**
+     * ID of the event. The semantics of this string are explicitly undefined to ease the implementation of producers. Enables deduplication.
+     */
+    String getEventID();
+
+    /**
+     * The version of the eventType. This enables the interpretation of data by eventual consumers, requires the consumer to be knowledgeable about the producer.
+     */
+    Optional<String> getEventTypeVersion();
+    /**
+     * Timestamp of when the event happened.
+     */
+    Optional<ZonedDateTime> getEventTime();
```

Comment:
Shouldn't this be an `Instant` instead of `ZonedDateTime`?

> This class models a single instantaneous point on the time-line. This might be used to record event time-stamps in the application.

➡️ https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html

Reply:
That was picked since the `Timestamp` (_String expression as defined in [RFC 3339](https://tools.ietf.org/html/rfc3339)_) can contain timezones.

I think that was suggested by @jroper on a mailing list thread a while ago